main.go

```diff
@@ -51,31 +52,53 @@ var (
 	enableLeaderElection = flag.Bool("leader-elect", false,
 		"Enable leader election for controller manager. "+
 			"Enabling this will ensure there is only one active controller manager.")
-	enableProfile = flag.Bool("enable-pprof", false, "enable pprof profiling")
-	profilePort   = flag.Int("pprof-port", 6060, "port for pprof profiling. defaulted to 6060 if unspecified")
 )
 
 func init() {
 	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
 
 	utilruntime.Must(eraserv1alpha1.AddToScheme(scheme))
-	//+kubebuilder:scaffold:scheme
 	utilruntime.Must(eraserv1.AddToScheme(scheme))
 	//+kubebuilder:scaffold:scheme
 }
 
 func main() {
+	var configFile string
+	flag.StringVar(&configFile, "config", "",
+		"The controller will load its initial configuration from this file. "+
+			"Omit this flag to use the default configuration values. "+
+			"Command-line flags override configuration from this file.")
 	flag.Parse()
 
 	if err := logger.Configure(); err != nil {
 		setupLog.Error(err, "unable to configure logger")
 		os.Exit(1)
 	}
 
-	if *enableProfile {
+	options := ctrl.Options{
+		Scheme:                 scheme,
+		MetricsBindAddress:     *metricsAddr,
+		Port:                   9443,
+		HealthProbeBindAddress: *probeAddr,
+		LeaderElection:         *enableLeaderElection,
```

Comment:
these should be updated to config?

Reply:
This isn't entirely clear from the code, but in line 89 these values will be overwritten if supplied in the config. I'm removing the flags and putting the defaults in here.

Reply:
done